src/LinearSolver.C

```diff
@@ -112,6 +113,10 @@ void TpetraLinearSolver::setupLinearSolver(
 
     // create the solver, e.g., gmres, cg, tfqmr, bicgstab
     LinSys::SolverFactory sFactory;
+    if (config_->get_method() == "TPETRA GMRES S-STEP") {
+      bool verbose = (params_->get<int>("Output Frequency", 0) > 0);
+      BelosTpetra::Impl::register_GmresSstep (verbose);
```

Comment:
We have to call a function with the option encoded into its name? Is that a static method? Is it putting something into a singleton? Why isn't the option just passed through params_ or however belos gets most of its option choices?

Reply:
This is not one of the Belos solvers. It uses Tpetra's MultiVector (instead of MvTraits). Currently, to use it through BelosSolverFactory, we need to register each solver. But, it will be nice to improve the interface. If you have any suggestions, please let us know.. Thank you.

Reply:
@iyamazaki Are you suggesting that there is a way to improve the interface on the Nalu-Wind side? If so, I'd recommend we take this opportunity to modify the solver registration interface so that we don't have to keep changing the solver interface and recompiling the code, as @jhux2 suggests.

Reply:
No.. I believe the changes need to happen in Belos side. We are discussing how to proceed..

Reply:
With [PR#7623](https://github.com/trilinos/Trilinos/pull/7623), we don't need to register GmresSstep. The code would compile with an earlier version of Trilinos, but will throw std::invalid_argument if we try to use sstep_gmres. Thank you so much for the review, and let us know if you have more comments!!